pkg/nfd-topology-gc/nfd-nrt-gc_test.go

```diff
@@ -22,212 +22,130 @@ import (
+		gc := newMockGC([]string{"node1", "node2"}, []string{"node1", "node2"})
+		// Override period to run fast
+		gc.gcPeriod = 100 * time.Millisecond
 
-		fakeClient := faketopologyv1alpha2.NewSimpleClientset(
-			&v1alpha2.NodeResourceTopology{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "node1",
-				},
-			},
-			&v1alpha2.NodeResourceTopology{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "node2",
-				},
+		nrt := v1alpha2.NodeResourceTopology{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "not-existing",
 			},
-		)
-
-		stopChan := make(chan struct{}, 1)
-
-		factory := informers.NewSharedInformerFactory(k8sClient, 5*time.Minute)
-		gc := &topologyGC{
-			factory:    factory,
-			topoClient: fakeClient,
-			stopChan:   stopChan,
-			gcPeriod:   10 * time.Minute,
 		}
 
-		err := gc.startNodeInformer()
-		So(err, ShouldBeNil)
+		errChan := make(chan error, 1)
+		go func() { errChan <- gc.Run() }()
 
-		nrts, err := fakeClient.TopologyV1alpha2().NodeResourceTopologies().List(context.TODO(), metav1.ListOptions{})
+		_, err := gc.topoClient.TopologyV1alpha2().NodeResourceTopologies().Create(context.TODO(), &nrt, metav1.CreateOptions{})
 		So(err, ShouldBeNil)
 
-		So(nrts.Items, ShouldHaveLength, 2)
-
-		err = k8sClient.CoreV1().Nodes().Delete(context.TODO(), "node1", metav1.DeleteOptions{})
-		So(err, ShouldBeNil)
-		// simple sleep with retry loop to make sure indexer will pick up event and trigger deleteNode Function
-		deleted := false
-		for i := 0; i < 5; i++ {
-			nrts, err := fakeClient.TopologyV1alpha2().NodeResourceTopologies().List(context.TODO(), metav1.ListOptions{})
-			So(err, ShouldBeNil)
-
-			if len(nrts.Items) == 1 {
-				deleted = true
-				break
-			}
-			time.Sleep(time.Second)
-		}
-		So(deleted, ShouldBeTrue)
+		So(waitForNRT(gc.topoClient, "node1", "node2"), ShouldBeTrue)
 	})
-	Convey("periodic GC should remove obsolete NRT", t, func() {
-		k8sClient := fakek8sclientset.NewSimpleClientset(
-			&corev1.Node{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "node1",
-				},
-			},
-			&corev1.Node{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "node2",
-				},
-			},
-		)
+}
 
-		fakeClient := faketopologyv1alpha2.NewSimpleClientset(
-			&v1alpha2.NodeResourceTopology{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "node1",
-				},
-			},
-			&v1alpha2.NodeResourceTopology{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "node2",
-				},
-			},
-		)
+func newMockGC(nodes, nrts []string) *mockGC {
+	k8sClient := fakek8sclientset.NewSimpleClientset(createFakeNodes(nodes...)...)
+	return &mockGC{
+		topologyGC: topologyGC{
+			factory:    informers.NewSharedInformerFactory(k8sClient, 5*time.Minute),
+			topoClient: faketopologyv1alpha2.NewSimpleClientset(createFakeNRTs(nrts...)...),
+			stopChan:   make(chan struct{}, 1),
+			gcPeriod:   10 * time.Minute,
+		},
+		k8sClient: k8sClient,
+	}
+}
 
-		stopChan := make(chan struct{}, 1)
+func createFakeNodes(names ...string) []runtime.Object {
```

Comment:
it would be more readable to return `[]*corev1.Node`, wdyt?

Reply:
It would (and I actually wrote it like that initially) but `NewSimpleClientset()` wants `runtime.Object` type so I changed it like this to avoid converting `[]*corev1.Node` to `[]runtime.Object` (plus the same conversion loop for NRTs).

Reply:
Aaa ok, slice to interface is a different type than interface. Still a bit rusty after PTO 😄

Reply:
Yes, that's the problem